Ülesanne 1: Arvutada tunnuste ja andmepunktide arv toorandmestikus

In [13]:
import pandas as pd

df = pd.read_csv("toorandmed.csv")

df.head(1)

course_uuid         code  \
0  eefb3a8d-7669-a5dc-9e13-6f95d719ebdf  LOMR.03.021   

                                       title__en  \
0  Practical Course of Microbiology and Virology   

                              version__title__en  credits  \
0  Practical Course of Microbiology and Virology      6.0   

                           overview__description__en  \
0  Gaining practical skills to work with microbes...   

                  version__overview__description__en  \
0  Gaining practical skills to work with microbes...   

                 overview__learning_outcomes_text_en  \
0  - After the course students are expected to:\r...   

                                   uuid state__code  ...  \
0  eefb3a8d-7669-a5dc-9e13-6f95d719ebdf   confirmed  ...   

  overview__description__es version__title__es  \
0                       NaN                NaN   

  version__overview__description__es title__de version__title__de  \
0                                NaN       NaN                NaN   

  overview__description__de title__vro overview__description__vro  \
0                       NaN        NaN                        NaN   

  version__title__vro version__overview__description__vro  
0                 NaN                                 NaN  

[1 rows x 223 columns]

- Andmepunktide arv (read):

In [14]:
len(df)


3031

Andmestikus on 3031 andmepunkti, kus iga punkt kirjeldab konkreetset kursust.

- Tunnuste arv (veerud):

In [15]:
len(df.columns)

223

Andmestikus on 223 tunnust (veergu) (tekst, numbrid, struktuursed väljad).

- Üks andmerida põhjalikult:

In [16]:
row = df.iloc[37]
row
for col, val in row.items():
    print(col, ":", str(val)[:100])



course_uuid : fb92a837-67a2-c588-7e58-09b4df26c6aa
code : P2OG.02.195
title__en : History and Theory of International Law
version__title__en : History and Theory of International Law
credits : 6.0
overview__description__en : International law cannot be understood only as contemporary body of norms. First of all, this body o
version__overview__description__en : International law cannot be understood only as contemporary body of norms. First of all, this body o
overview__learning_outcomes_text_en : - The student who has passed the course is able to:
- -- Understand the main periods and central fi
uuid : fb92a837-67a2-c588-7e58-09b4df26c6aa
state__code : confirmed
state__et : Kinnitatud
state__en : confirmed
last_update : 2024-02-05T11:04:36
title__et : Rahvusvahelise õiguse ajalugu ja teooria
general__input_languages : [{"code": "et", "en": "Estonian", "et": "eesti keel"}]
general__structural_unit_shares : [{"coordinating": true, "course_share_percent": 100, "structural_unit_code": "SVOI

*Tähelepanekud*: Andmestiku tunnused on veergudena struktureeritud ning kasutavad nimetustes kahekordset alakriipsu __. Samuti on ühes tabelis esitatud sama info mitmes keeles (ka kursuse üldinfo ja versioonipõhine info). Andmetüübid on erinevad: leidub nii tekstilisi väärtusi (kirjeldused, nimetused), numbrilisi väärtusi (credits) kui ka identifikaatoreid (uuid). Puuduvad väärtused on esitatud NaN-ina, eriti keelepõhistes veergudes, mis tähendab, et kõigi keelte jaoks infot alati olemas pole. Mõned tunnuseid on dubleerivad või väga sarnased (title__en ja version__title__en).

### Kokkuvõte
Andmestik sisaldab 3031 andmepunkti ja 223 tunnust.
Üks andmerida kirjeldab ühte ülikooli ainet koos metaandmete,
õpiväljundite, hindamise ja struktuuriinfoga.
Andmed on keerukad, mitmekeelsed.